pkg/kncloudevents/message_sender.go

```diff
@@ -160,6 +160,51 @@ func RetryConfigFromDeliverySpec(spec duckv1.DeliverySpec) (RetryConfig, error)
 	return retryConfig, nil
 }
 
-func checkRetry(_ context.Context, resp *nethttp.Response, err error) (bool, error) {
-	return !(resp != nil && resp.StatusCode < 300), err
+// Simple default implementation
+func RetryIfGreaterThan300(_ context.Context, response *nethttp.Response, err error) (bool, error) {
+	return !(response != nil && (response.StatusCode < 300)), err
+}
+
+// Alternative function to determine whether to retry based on response
+//
+// Note - Returning true indicates a retry should occur.  Returning an error will result in that
+//        error being returned instead of any errors from the Request.
+//
+func SelectiveRetry(_ context.Context, response *nethttp.Response, err error) (bool, error) {
```

Comment:
I think It would be great if we summarize which codes are retried and which are not in the function comment. I know they're summarized below but it wont show up in the pkg generated docs.